### DISCUSSION TITLE
Flatten names in Strings resources structured naming

### DISCUSSION THREAD

**Discussion by biappi** [SwiftGen/SwiftGen#902] [Ideas]
Hello!

Toying with SwiftGen while evaluating for a project, I implemented a simple method to flatten the name spacing created by the structured strings template.

For example, having two keys `an_example_key` and `an_example_value` currently creates enums accessed by `An.Example.Key` and `An.Example.Value`. The implementation groups together the enums with a single element, to produce an  output like `AnExample.Key` and `AnExample.Value`.

I have a [rough patch](https://github.com/biappi/SwiftGen/blob/9e1241a00b9439cc9c72786641db862636f8a126/Sources/SwiftGenKit/Stencil/StringsParser%2BContext.swift) that I can clean up if there is interest / a PR would be accepted.

Thanks

**Comment by djbe** [comment]
Might be better if this could be achieved at the template level, instead of messing with the context itself, as that would obviously be a massive breaking change for all our users.
